### PR TITLE
Story 5 updated

### PR DESCRIPTION
**Pull request by Crutovel**
Fixed story 5 in accordance with the requirements.
This code needs some potential improvements in flexibility. 
Btw docs are to be provided later on. 
If you want to test use this json:
{
  "locations": [ 1,2]
}

### CODE REVIEW

File: src/main/java/com/softserve/teamproject/dto/GroupsFilterRequestWrapper.java [hunk 1]
```diff
@@ -0,0 +1,14 @@
+package com.softserve.teamproject.dto;
+
+public class GroupsFilterRequestWrapper {
```
Comment (svoronin) [inline src/main/java/com/softserve/teamproject/dto/GroupsFilterRequestWrapper.java]: I think you can drop second half of the class name without any loss of context.

File: src/main/java/com/softserve/teamproject/service/impl/GroupServiceImpl.java [hunk 2]
```diff
@@ -105,4 +88,28 @@ public void deleteGroup(int groupId, String userName) {
...
+
+    if (wrapper.getLocations() != null) {
+      predicates.add(GroupExpressions.getByLocationIds(wrapper.getLocations()));
+    }
+
+    if (predicates.size() > 0) {
+      BooleanExpression predicate = predicates.get(0);
+
+      for (int i = 1; i < predicates.size(); i++) {
+        predicate = predicate.or(predicates.get(i));
```
Comment (svoronin) [inline src/main/java/com/softserve/teamproject/service/impl/GroupServiceImpl.java]: That code shouldn't be reachable. You are putting a single item in list on line 102. I can't see where you could get more that one entry in the list. So I feel like this code should be optimized. Please recheck this.

File: src/main/java/com/softserve/teamproject/dto/GroupsFilterRequestWrapper.java (hunk 1 above)
Comment (Crutovel) [inline src/main/java/com/softserve/teamproject/dto/GroupsFilterRequestWrapper.java]: I agree

File: src/main/java/com/softserve/teamproject/service/impl/GroupServiceImpl.java (hunk 2 above)
Comment (Crutovel) [inline src/main/java/com/softserve/teamproject/service/impl/GroupServiceImpl.java]: future-coding stuff is bad practice. Thank you, I ll fix this.

File: src/main/java/com/softserve/teamproject/controller/GroupController.java [hunk 3]
```diff
@@ -79,7 +79,7 @@ public void deleteGroup(@PathVariable Integer id, Principal principal) {
    * @return groups info
    */
   @RequestMapping(value = "/groups/filter", method = RequestMethod.POST)
-  public Iterable getGroupsByFilter(@RequestBody GroupsFilterRequestWrapper requestWrapper) {
+  public Iterable getGroupsByFilter(@RequestBody GroupsFilter requestWrapper) {
```
Comment (svoronin) [inline src/main/java/com/softserve/teamproject/controller/GroupController.java]: I'd suggest renaming variables here and in the other occurrences from wrapper to something filter-related

Comment (Crutovel) [inline src/main/java/com/softserve/teamproject/controller/GroupController.java]: Done.

**Comment by svoronin** [comment]
LGTM. Please resolve merge conflicts and it should be ready for final review.

**Comment by khdimon** [comment]
When we send incorect JSON (for example empty { })to groups/filter we take 200 OK and emtry response. Maybe error could be better.

**Pull request merged**